### PR TITLE
Switch to `System.CommandLine` for tools

### PR DESCRIPTION
**Pull request by rubo**
See #7653. The changes applied as a separate PR for easier review.

## Changes

- **DocGen**: Replaced `Spectre.Console.Cli` with `System.CommandLine`
- **HiveCompare**: Replaced `McMaster.Extensions.CommandLineUtils` with `System.CommandLine`
- **Kute**: Replaced `CommandLineParser` with `System.CommandLine`
- **SendBlobs**: Replaced `McMaster.Extensions.CommandLineUtils` with `System.CommandLine`
- **Nethermind.Cli**: Replaced `McMaster.Extensions.CommandLineUtils` with `System.CommandLine`

**No API or CLI usage changes**

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [x] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

### CODE REVIEW

File: tools/Nethermind.Tools.Kute/ResponseTracer/FileResponseTracer.cs [hunk 1]
```diff
@@ -7,9 +7,9 @@ namespace Nethermind.Tools.Kute.ResponseTracer;
 
 public class FileResponseTracer : IResponseTracer
 {
-    private readonly string _tracesFilePath;
+    private readonly string? _tracesFilePath;
 
-    public FileResponseTracer(string tracesFilePath)
+    public FileResponseTracer(string? tracesFilePath)
```
Comment (emlautarom1) [inline tools/Nethermind.Tools.Kute/ResponseTracer/FileResponseTracer.cs:12]: This parameter should never be null

File: tools/DocGen/Properties/launchSettings.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "profiles": {
     "DocGen": {
       "commandName": "Project",
-      "commandLineArgs": "/path/to/docs --config --dbsize --jsonrpc --metrics"
+      "commandLineArgs": "/path/to/docs --config --jsonrpc --metrics"
```
Comment (emlautarom1) [inline tools/DocGen/Properties/launchSettings.json:5]: Why are we removing the `--dbsize` flag?

File: tools/Nethermind.Tools.Kute/Config.cs [hunk 3]
```diff
@@ -1,135 +1,80 @@
...
 
-public class Config
+public static class Config
 {
-    [Option(
-        shortName: 'i',
-        longName: "input",
+    public static CliOption<string> MessagesFilePath { get; } = new("--input", "-i")
+    {
+        Description = "Path to a file containing JSON RPC messages",
```
Comment (emlautarom1) [inline tools/Nethermind.Tools.Kute/Config.cs:13]:
```suggestion
        Description = "Path to a file or directory containing JSON RPC messages",
```

File: tools/Nethermind.Tools.Kute/Config.cs [hunk 4]
```diff
@@ -1,135 +1,80 @@
...
-        shortName: 'u',
-        longName: "unwrapBatch",
-        Required = false,
-        Default = false,
-        HelpText = "If true then each batched request will be unwraped to single requests."
-    )]
-    public bool UnwrapBatch { get; }
+    public static CliOption<int> RequestsPerSecond { get; } = new("--rps", "-e")
+    {
+        Description = "If set to higher than 0, then requests will be send in selected RPS (Requests per seconds) rate. If 0 (or lower) then requests will be sent sequentionally.",
```
Comment (emlautarom1) [inline tools/Nethermind.Tools.Kute/Config.cs:72]:
```suggestion
        Description = "If set to higher than 0, then requests will be send in selected RPS (Requests per seconds) rate. If 0 (or lower) then requests will be sent sequentially",
```

File: tools/Nethermind.Tools.Kute/Config.cs [hunk 5]
```diff
@@ -1,135 +1,80 @@
...
-        AuthTtl = authTtl;
-        DryRun = dryRun;
-        ShowProgress = showProgress;
-        MetricsOutputFormatter = metricsOutputFormatter;
-        MethodFilters = methodFilters;
-        ResponsesTraceFile = responsesTraceFile;
-        RequestsPerSecond = requestsPerSecond;
-        UnwrapBatch = unwrapBatch;
-    }
+        Description = "If true then each batched request will be unwraped to single requests."
```
Comment (emlautarom1) [inline tools/Nethermind.Tools.Kute/Config.cs:78]:
```suggestion
        Description = "If true then each batched request will be unwraped to single requests"
```

File: tools/Nethermind.Tools.Kute/Program.cs [hunk 6]
```diff
@@ -15,73 +14,92 @@
...
+
+        return await cli.InvokeAsync(args);
     }
 
-    static IServiceProvider BuildServiceProvider(Config config)
+    private static IServiceProvider BuildServiceProvider(ParseResult parseResult)
     {
+        bool dryRun = parseResult.GetValue(Config.DryRun);
+        bool unwrapBatch = parseResult.GetValue(Config.UnwrapBatch);
+        string? responsesTraceFile = parseResult.GetValue(Config.ResponsesTraceFile);
```
Comment (emlautarom1) [inline tools/Nethermind.Tools.Kute/Program.cs:56]: I think we can inline these variables, no need to have them defined here.

File: tools/SendBlobs/SetupCli.cs [hunk 7]
```diff
@@ -1,84 +1,107 @@
...
-        app.OnExecuteAsync(async cancellationToken =>
+        CliOption<string> rpcUrlOption = new("--rpcurl")
+        {
+            Description = "The URL of the JSON RPC server",
+            HelpName = "URL",
+            Required = true
+        };
+        CliOption<string> blobTxOption = new("--bloboptions")
+        {
+            Description = "Options in format '10x1-2', '2x5-5' etc. for the blobs.",
```
Comment (emlautarom1) [inline tools/SendBlobs/SetupCli.cs:26]:
```suggestion
            Description = "Options in format '10x1-2', '2x5-5' etc. for the blobs",
```

File: tools/nuget.config [hunk 8]
```diff
@@ -12,5 +13,9 @@
     <packageSource key="system.commandline">
       <package pattern="System.CommandLine*" />
     </packageSource>
+    <!--<packageSource key="nugettest.org">
+      <package pattern="" />
+    </packageSource>
+  </packageSourceMapping>-->
```
Comment (emlautarom1) [inline tools/nuget.config:19]: Commented out code

File: tools/nuget.config [hunk 9]
```diff
@@ -4,6 +4,7 @@
     <clear />
     <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
     <add key="system.commandline" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-libraries/nuget/v3/index.json" />
+    <!--<add key="nugettest.org" value="https://apiint.nugettest.org/v3/index.json" />-->
```
Comment (emlautarom1) [inline tools/nuget.config:7]: Commented out code

**Review by emlautarom1 — APPROVED** [review]
Some small details but LGTM

File: tools/nuget.config (hunk 9 above)
Comment (rubo) [reply tools/nuget.config:7]: This is intentional for convenience, as we sometimes use the test feed.

File: tools/DocGen/Properties/launchSettings.json (hunk 2 above)
Comment (rubo) [reply tools/DocGen/Properties/launchSettings.json:5]: We don't. It's for local launch, where it doesn't work without actual data.

File: tools/Nethermind.Tools.Kute/ResponseTracer/FileResponseTracer.cs (hunk 1 above)
Comment (rubo) [reply tools/Nethermind.Tools.Kute/ResponseTracer/FileResponseTracer.cs:12]: In this case, I will make it required so the app will throw when the flag is missing.

File: tools/Nethermind.Tools.Kute/Program.cs (hunk 6 above)
Comment (rubo) [reply tools/Nethermind.Tools.Kute/Program.cs:56]: They are being used in multiple places, so it fetches their values once.

File: tools/Nethermind.Tools.Kute/ResponseTracer/FileResponseTracer.cs [hunk 10]
```diff
@@ -5,20 +5,15 @@
 
 namespace Nethermind.Tools.Kute.ResponseTracer;
 
-public class FileResponseTracer : IResponseTracer
+public class FileResponseTracer(string tracesFilePath) : IResponseTracer
```
Comment (emlautarom1) [inline tools/Nethermind.Tools.Kute/ResponseTracer/FileResponseTracer.cs:8]: No need to change this file at all: null check is performed in `Program.cs` and `tracesFilePath` is never null.

**Pull request merged**